lib/Cro/HTTP/ReverseProxy.pm6

```diff
@@ -0,0 +1,91 @@
+use Cro;
+use Cro::HTTP::Client;
+
+class Cro::HTTP::ReverseProxy does Cro::Transform {
+    has $.to;
+    has $.to-absolute;
+    has $!destination;
+    has $!client;
+    has $.request;
+    has $.response;
+    has %.ca;
+
+    submethod TWEAK() {
+        unless ($!to.defined ^^ $!to-absolute.defined) {
+            die 'Either `to` or `to-absolute` must be specified for ReverseProxy';
+        }
+        $!to = self!trim-url($!to) if $!to ~~ Str;
+        $!to-absolute = self!trim-url($!to-absolute) if $!to-absolute ~~ Str;
+        with $!to {
+            die '`to` of ReverseProxy must be a Str or code block' unless $!to ~~ Str|Callable;
+            $!destination = $!to;
+        }
+        else {
+            $!destination = $!to-absolute;
+        }
+
+        my %args = %!ca ?? (:%!ca, :http<1.1 2>) !! ();
+        $!client = Cro::HTTP::Client.new(|%args);
+    }
+
+    method consumes() { Cro::HTTP::Request }
+    method produces() { Cro::HTTP::Response }
+
+    method !trim-url($url) { $url.ends-with('/') ?? $url.substr(0, *-1) !! $url }
+
+    method transformer(Supply $pipeline --> Supply) {
+        supply whenever $pipeline -> $request {
+            my $*PROXY-STATE;
+            sub send($request) {
+                my %options = headers => $request.headers,
+                              body => await $request.body if $request.has-body;
+
+                whenever self!make-destination($request) {
+                    my $response = await $!client.request($request.method, $_, %options);
```

Comment:
We're mixing `whenever` and `await` here and it's not entirely clear to me we want to be. `await` blocks the request pipeline, so I think if we had other HTTP/2 streams we'd not be able to concurrently handle the proxying of the requests so efficiently with the `await`.